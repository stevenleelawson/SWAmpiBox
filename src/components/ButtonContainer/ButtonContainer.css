.button-container {
  height: 30vh;
  width: 100%;
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
}
