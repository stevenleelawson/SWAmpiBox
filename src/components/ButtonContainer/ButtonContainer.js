import React from 'react';
import Button from '../Button/Button';
import './ButtonContainer.css';

const ButtonContainer = () => {
  return (
    <div className='button-container'>
      <Button />
      <Button />
      <Button />
    </div>
  )
}
export default ButtonContainer;
