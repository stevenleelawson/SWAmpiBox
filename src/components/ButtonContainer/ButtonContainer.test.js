import React from 'react';
import { shallow } from 'enzyme';
import ButtonContainer from './ButtonContainer';

describe('ButtonContainer', () => {
  it('should match the snapshot', () => {
    const mockFavorites = {
      name: 'Luke',
      species: 'human',
      homeworld: 'tattooine'
    }
    const wrapper = shallow(<ButtonContainer favorites={[mockFavorites]}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
