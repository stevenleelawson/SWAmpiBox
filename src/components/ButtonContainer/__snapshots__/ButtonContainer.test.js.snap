// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer
    favorites={
        Array [
            Object {
              "homeworld": "tattooine",
              "name": "Luke",
              "species": "human",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/people"
>
          People
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/vehicles"
>
          Vehicles
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/planets"
>
          Planets
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          to="/favorites"
>
          Favorites 
          <p>
                    1
          </p>
</NavLink>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "People",
          "className": "button",
          "onClick": [Function],
          "to": "/people",
        },
        "ref": null,
        "rendered": "People",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Vehicles",
          "className": "button",
          "onClick": [Function],
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Planets",
          "className": "button",
          "onClick": [Function],
          "to": "/planets",
        },
        "ref": null,
        "rendered": "Planets",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": Array [
            "Favorites ",
            <p>
              1
</p>,
          ],
          "className": "button",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Array [
          "Favorites ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "p",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/people"
>
            People
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/vehicles"
>
            Vehicles
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/planets"
>
            Planets
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            to="/favorites"
>
            Favorites 
            <p>
                        1
            </p>
</NavLink>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "className": "button",
            "onClick": [Function],
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "className": "button",
            "onClick": [Function],
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "className": "button",
            "onClick": [Function],
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "Favorites ",
              <p>
                1
</p>,
            ],
            "className": "button",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
